trin-core/src/utils/db.rs

```diff
@@ -1,30 +1,64 @@
 use std::{env, fs};
 
+use crate::cli::TrinConfig;
 use directories::ProjectDirs;
 use discv5::enr::NodeId;
 use rocksdb::{Options, DB};
+use std::path::PathBuf;
+use std::time::SystemTime;
 
 const TRIN_DATA_ENV_VAR: &str = "TRIN_DATA_PATH";
 
 pub fn get_data_dir(node_id: NodeId) -> String {
-    let path = env::var(TRIN_DATA_ENV_VAR).unwrap_or_else(|_| get_default_data_dir(node_id));
+    let application_path = env::var(TRIN_DATA_ENV_VAR).unwrap_or_else(|_| get_default_data_dir());
+    let trin_config = TrinConfig::from_cli();
 
-    fs::create_dir_all(&path).expect("Unable to create data directory folder");
-    path
-}
-
-pub fn get_default_data_dir(node_id: NodeId) -> String {
-    // Windows: C:\Users\Username\AppData\Roaming\Trin\data
-    // macOS: ~/Library/Application Support/Trin
-    // Unix-like: $HOME/.local/share/trin
+    fs::create_dir_all(&application_path).expect("Unable to create data directory folder");
 
     // Append first 8 characters of Node ID
-    let mut application_string = "Trin_".to_owned();
     let node_id_string = hex::encode(node_id.raw());
     let suffix = &node_id_string[..8];
-    application_string.push_str(suffix);
+    let path = format!("{}/Trin_{}", &application_path, &suffix);
+
+    if trin_config.ephemeral {
+        fs::create_dir_all(&path).expect("Unable to create ephemeral data directory");
+        let paths = fs::read_dir(&application_path).unwrap();
+        // NOTE: SystemTime is machine dependent
+        let mut min_timestamp = SystemTime::now();
+        let mut min_path = PathBuf::new();
+        let mut n_paths = 0;
+        for p in paths {
```

Comment:
*minor nitpick* - we try to stay away from single-letter variables - I'd suggest going with `path` for `p` (and updating the prior `path` on line 21 to something else) - as well as `path_count` / `num_paths` for `n_paths`. In general avoiding single letter vars helps a ton with readability which is a priority over minimizing code length, especially in open-source when you can expect a fair amount of different people to be reading this code.